### PR TITLE
Add's the SA Ronin .20 Handgun

### PR DESCRIPTION
**Pull request by thedarkcrusader**
s

## About The Pull Request
This PR Add's the **SA Ronin .20 Handgun** and the disks and ammo to the respective places the Ronin is a firearm with the same rarity as the Dalas and is meant to be a Dalas equivalent for handguns. its role isn't to do much direct damage but its a hit and run firearm that is able to reliably penetrate armor. and get away with it and as a bonus the firearm area Denial potential with an integrated flamethrower in its barrel that spews out a 3-tile horizontal flame Infront of the user that ignites anyone when they spin there firearm.

Thanks to Humon. Kirov, Baron, Mycah . Garen and anyone else im forgetting

## Why It's Good For The Game
an interesting firearm with good potential for harassment i think would be a welcome addition 

## Testing
Everything spawns as it should there needs to be a bit if in game testing however. 

## Changelog
:cl:
add: Adds the Ronin, adds mags for it, Add sprites for it
/:cl:

### CODE REVIEW

**Comment by DiaFRAME444** [comment]
There should be a note about the integrated spin-flamethrower in its description, I think.

**Comment by thedarkcrusader** [comment]
> There should be a note about the integrated spin-flamethrower in its description, I think.

yeah ill do that

**Comment by hyperioo** [comment]
There is a conflict with the ammo mag dmi because #8193 added new magazines 3 weeks ago and your master branch has not been updated.

**Comment by Firefox13** [comment]
I asked you and Tristain several times to talk with me so we could discuss its proper capabilities.
I'm not against the gun but since I didn't de-facto approved it I'm blocking this PR *until* it gets proper testing and balancing.

File: code/modules/projectiles/guns/projectile/pistol/ronin.dm [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	fire_sound = 'sound/weapons/guns/fire/revolver_fire.ogg'
+	origin_tech = list(TECH_COMBAT = 2, TECH_MATERIAL = 2)
+	matter = list(MATERIAL_PLASTEEL = 12, MATERIAL_PLASTIC = 6)
+	price_tag = 5000 //On par with the Dallas in terms of rarity
+	caliber = CAL_SRIFLE
+	ammo_type = /obj/item/ammo_casing/srifle
+	load_method = MAGAZINE
+	mag_well = MAG_WELL_PISTOL
+	magazine_type = /obj/item/ammo_magazine/srifle/short
+	proj_step_multiplier = 1.2
```
Comment (Firefox13) [inline code/modules/projectiles/guns/projectile/pistol/ronin.dm:19]: Higher than 1 = LOWER bullet speed. Is this what you want?

File: code/modules/projectiles/guns/projectile/pistol/ronin.dm [hunk 2]
```diff
@@ -0,0 +1,71 @@
+/obj/item/gun/projectile/ronin
+	name = "SA HG .20 \"Ronin\""
+	desc = " A handgun which was designed with hit and run tactics in mind, not the most powerful firearm but its ability to reliably penetrate armor and ricochet rounds of walls make it a choice for celebrities. and playboys looking to be a nuisance..."
+	icon = 'icons/obj/guns/projectile/ronin.dmi'
+	icon_state = "silverhand"
+	item_state = "silverhand"
+	w_class = ITEM_SIZE_NORMAL
+	can_dual = TRUE
+	silenced = TRUE
```
Comment (Firefox13) [inline code/modules/projectiles/guns/projectile/pistol/ronin.dm:9]: They definitely do not come silenced.
https://youtu.be/rG9_I81dcZI

**Comment by Firefox13** [comment]
Credit people in changelog properly.

File: code/modules/projectiles/guns/projectile/pistol/ronin.dm [hunk 3]
```diff
@@ -0,0 +1,71 @@
+/obj/item/gun/projectile/ronin
+	name = "SA HG .20 \"Ronin\""
+	desc = " A handgun which was designed with hit and run tactics in mind, not the most powerful firearm but its ability to reliably penetrate armor and ricochet rounds of walls make it a choice for celebrities. and playboys looking to be a nuisance..."
```
[the remark this answers is not in the document]
Comment (DiaFRAME444) [reply code/modules/projectiles/guns/projectile/pistol/ronin.dm:3]: "that allows bullets to ricochet of the walls"
Off the walls.

[the remark this answers is not in the document]
Comment (Firefox13) [reply code/modules/projectiles/guns/projectile/pistol/ronin.dm:3]: Fixed, thanks.

File: code/modules/projectiles/ammunition/magazines.dm [hunk 4]
```diff
@@ -226,6 +226,42 @@
...
+	icon_state = "ronin"
+	origin_tech = list(TECH_COMBAT = 2)
+	mag_type = MAGAZINE
+	mag_well = MAG_WELL_PISTOL
+	caliber = CAL_SRIFLE
+	matter = list(MATERIAL_STEEL = 9)
+	ammo_type = /obj/item/ammo_casing/srifle
+	max_ammo = 10
+	ammo_states = list(10)
+	rarity_value = 20
```
[the remark this answers is not in the document]
Comment (thedarkcrusader) [reply code/modules/projectiles/ammunition/magazines.dm:246]: they do sometimes

**Comment by Firefox13** [comment]
While we will be testing the gun can you please add the jacket?

**Comment by thedarkcrusader** [comment]
Sure sure expect a pr around tomorrow or Monday

**Comment by thedarkcrusader** [comment]
dont you love conflicts i sure do

**Comment by thevandie** [comment]
CUMSHOE, FIX IT
FIREFOX, MERGE IT

**Comment by thedarkcrusader** [comment]
I'll fix this this week but I currently don't like how it's balanced right now it still feels underwhelming even with the flamethrower inside it's barrel

**Comment by thedarkcrusader** [comment]
And considering it's supposed to be as rare as the Dallas it's in need of a major tune up

**Comment by thevandie** [comment]
👍

**Comment by thedarkcrusader** [comment]
Going  to rebalance this now

File: code/modules/projectiles/ammunition/magazines.dm (hunk 4 above)
[the remark this answers is not in the document]
Comment (Humonitarian) [reply code/modules/projectiles/ammunition/magazines.dm:246]: > Do we want them just lying around? Do Dallas magazines spawn by themselves? I really don't remember.

Yes.

**Comment by thedarkcrusader** [comment]
uhhh this was not suppused to happen

**Pull request closed without merging**